hips/hip-9999.md

```diff
@@ -0,0 +1,79 @@
+
+## Specification
+
+Nothing would change from the user's perspective. With some modifications in the current code it's possible to add this metadata into the release object transparently.
+
+The current idea for this implementation is to add a field in both the `ChartDownloader` and `Metadata` (in the `pkg/chart/metadata.go` file) structs to hold the repository URL. The Downloader is the logical place to derive the proper URL and can then that storage location can populate the Metadata field.
+
+More specifically, the URL is included in the local repository index file for a locally named repository. For instance, running `helm repo add [NAME] [URL]` would add the repository to the index file with the `[URL]` passed as the real location to download charts. In the helm code, the downloader reads this file to find the URL which is what gets bubbled up to the release object.
+
+If for any reason this data can not be gathered at install/upgrade time, then that particular field would be omitted from the release object metadata.
+
+The chart metadata object would have new fields that looked like so:
+
+```
+"repoURL": "https://charts.fairwinds.com/stable"
+"tarballDigest": "sha256:f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9"
+```
+
+And an OCI compliant repository would look like this (example is using ECR):
+
+```
+"repoURL": "oci://${ACCOUNT_NUMBER}.dkr.ecr.us-east-1.amazonaws.com/goldilocks"
+"tarballDigest": "sha256:f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9"
+```
+
+## Backwards compatibility
+
+There should be no impact to backwards compatibility.
+
+## Security implications
+
+A potential security issue would be private repository URLs becoming present in a cluster via this implementation. A potential fix for this would be to add a command line argument that would keep this field in the release object empty, but that would introduce something a user has to change in their workflow which may not be desirable. Another way would be to attempt a DNS match against a publicly available DNS server and if it's not found, then we do not populate the repository URL field.
+
+Another option is to ignore this potential downside. Unclear if this would truly be an issue or not. Would love to have feedback on why it would be bad for an internal repository URL to be stored in the cluster.
+
+## How to teach this
+
+The feature can be documented by describing how we get the URL (taken from the description in the [Specification](#Specification) section). The usage of this URL is up to a user or third-party tool developer, but the URL would be included in the release object as one more informational piece with no prescribed usage.
+
+## Reference implementation
+A PR has been created to show a possible implementation: https://github.com/helm/helm/pull/10369
```

Comment:
The reference implementation, as far as I can tell, does not do anything with the `tarballDigest` field, which is probably the correct behavior for now (given the note I left on the Security Implications)

Reply:
This is correct, I have not done that work yet.